pandas/lib.pyx

```diff
@@ -1055,6 +1055,20 @@ def string_array_replace_from_nan_rep(ndarray[object, ndim=1] arr, object nan_re
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+def object_array_decode_bytes(ndarray[object, ndim=1] arr, object encoding):
+    """Decode any instances of bytes to str in arr using the given encoding."""
```

Comment:
@jzwinck pls work on the location / logic first. Most things already exist in pandas. We are very DRY, so move it to ``.to_native_dtype`` as I indicated. And don't write new cython, until/unless it becomes an issue (unlikely).